samplePDF/samplePDFFDBase.h

```diff
@@ -238,18 +231,40 @@ class samplePDFFDBase :  public samplePDFBase
   //===========================================================================
   //DB Vectors to store which kinematic cuts we apply
   //like in XsecNorms but for events in sample. Read in from sample yaml file 
-  std::vector< std::string > SelectionStr; 
-  std::vector< std::vector<double> > Selection; //The enum, then the bounds corresponding to the string
-
-  // like in XsecNorms but for events in sample. Read in from sample yaml file
-  // in samplePDFExperimentBase.cpp
-  std::vector< std::vector<double> > SelectionBounds;
-
   //What gets used in IsEventSelected, which gets set equal to user input plus 
   //all the vectors in StoreSelection
-  //std::vector< std::vector<double> > Selection;
+  /// @brief the Number of selections in the 
   int NSelections;
-  //What gets pulled from config options
-  std::vector< std::vector<double> > StoredSelection; 
-  //===========================================================================
+  
+  /// @brief What gets pulled from config options, these are constant after loading in
+  /// this is of length 3: 0th index is the value, 1st is lower bound, 2nd is upper bound
+  std::vector< std::vector<double> > StoredSelection;
+  /// @brief the strings grabbed from the sample config specifying the selections
+  std::vector< std::string > SelectionStr; 
+  /// @brief the bounds for each selection lower and upper
+  std::vector< std::vector<double> > SelectionBounds;
+  /// @brief a way to store selection cuts which you may push back in the get1DVar functions
+  /// most of the time this is just the same as StoredSelection
+  std::vector< std::vector<double> > Selection;
+   //===========================================================================
+
+  manager* SampleManager;
+  void InitialiseSingleFDMCObject(int iSample, int nEvents);
```

Comment:
I don't have strong opinion but do we need manager* or YAML::NODE woulbe be enough?

Reply:
manager would be better because we use CheckNodeExists functions: https://github.com/mach3-software/MaCh3/blob/5fa87ded750b3ba2f040e7eddd9e51e7dcc7350b/samplePDF/samplePDFFDBase.cpp#L56

Reply:
Check node exist isn't part of manager but YamlHelper.h
https://github.com/mach3-software/MaCh3/blob/60dfdff0c2e1c709161779eb5d1f23b2e280a3b4/manager/YamlHelper.h#L56

So you can just pass yaml node.

Again I don't have strong opinions about this one

Reply:
Ah okay, seems like there's no strong reason to move it to a YAML::Node

Reply:
The reason is that people can look up documentation for yaml-cpp and work out how to interact with the objects, if you wrap everything, either you have to supply your own wrapped interface to the library features, or you hide possibly useful features of the library.

Reply:
Plus, it'll take probably 42 seconds with find/replace to change it